### PR TITLE
fix: updates arc ext check

### PR DESCRIPTION
**Pull request by nilekhc**
Signed-off-by: Nilekh Chaudhari <1626598+nilekhc@users.noreply.github.com>

**Reason for Change**:

**Requirements**

- [x] squashed commits
- [ ] included documentation
- [ ] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

fixes: https://github.com/Azure/secrets-store-csi-driver-provider-azure/issues/892

**Does this change contain code from or inspired by another project?**

- [ ] Yes
- [x] No

**If "Yes," did you notify that project's maintainers and provide attribution?**

**Special Notes for Reviewers**:

### CODE REVIEW

**Comment by nilekhc** [comment]
cc @vittoriocanilli

**Review by vittoriocanilli — APPROVED** [review]
LGTM.

I am not sure if you could have fixed it as well, by simply swapping the operators in the AND condition instead of having 2 nested IF statements.

Anyway I am not familiar with writing Helm charts 😄

**Comment by nilekhc** [comment]
> Helm uses go template and the documentation (https://pkg.go.dev/text/template#hdr-Functions) does confirm that the arguments are evaluated from left to right. So if the first one is false, it should return false as the result is determined.
> 
> Could we switch the order like @vittoriocanilli suggested instead of nested ifs?

That's better. Thanks for the suggestion @vittoriocanilli

File: manifest_staging/charts/csi-secrets-store-provider-azure/templates/provider-azure-installer.yaml [hunk 1]
```diff
@@ -22,14 +22,16 @@ spec:
 {{- if .Values.linux.podAnnotations}}
 {{- toYaml .Values.linux.podAnnotations | nindent 8 }}
 {{- end }}
-{{- if and .Values.arc.enableMonitoring .Values.enableArcExtension }}
+{{- if and .Values.enableArcExtension .Values.arc.enableMonitoring }}
         prometheus.io/scrape: "true" 
         prometheus.io/port: "8898"
 {{- end }}
+{{- end }}
```
Comment (aramase) [inline manifest_staging/charts/csi-secrets-store-provider-azure/templates/provider-azure-installer.yaml:29]:
```suggestion
```

File: manifest_staging/charts/csi-secrets-store-provider-azure/templates/provider-azure-installer.yaml [hunk 2]
```diff
@@ -22,14 +22,16 @@ spec:
...
 # this will ensure pods are rolled in next helm upgrade if cert files are updated.
-{{- if and .Values.Azure.proxySettings.isProxyEnabled .Values.enableArcExtension }}
+{{- if and .Values.enableArcExtension .Values.Azure.proxySettings.isProxyEnabled }}
         checksum/arc-proxy-config: {{ include (print $.Template.BasePath "/arc-proxy-secret.yaml") . | sha256sum }}
 {{- end }}
+{{- end }}
```
Comment (aramase) [inline manifest_staging/charts/csi-secrets-store-provider-azure/templates/provider-azure-installer.yaml:34]:
```suggestion
```

**Review by aramase — APPROVED** [review]

**Pull request merged**